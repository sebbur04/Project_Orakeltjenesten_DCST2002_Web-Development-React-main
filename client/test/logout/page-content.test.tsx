// Import necessary modules for testing
import * as React from "react";
import { shallow } from "enzyme";
import PageContent from "../../src/logout/page-content";

jest.mock("../../src/service/versionlog-service", () => ({
  getLatestVersion: jest.fn(() =>
    Promise.resolve({
      id: 1,
      content: "## Sample Content",
      userid: 1,
      date: "2024-01-01T12:00:00Z",
      changelog: "Initial version",
      version: 1,
      pageid: 1,
    })
  ),
}));

jest.mock("../../src/service/page-service", () => ({
  getPage: jest.fn(() =>
    Promise.resolve({
      id: 1,
      name: "Sample Page",
      num_views: 123,
    })
  ),
  getTagsFromPage: jest.fn(() =>
    Promise.resolve([
      { id: 1, name: "Tag1" },
      { id: 2, name: "Tag2" },
    ])
  ),
}));

jest.mock("../../src/service/comment-service", () => ({
  getCommentsFromPage: jest.fn(() =>
    Promise.resolve([
      {
        id: 1,
        content: "Great post!",
        userid: 1,
        date: "2024-01-01T12:30:00Z",
      },
      {
        id: 2,
        content: "Needs more details.",
        userid: 2,
        date: "2024-01-01T13:00:00Z",
      },
    ])
  ),
}));

jest.mock("../../src/service/user-service", () => ({
  getAllUsers: jest.fn(() =>
    Promise.resolve([
      { id: 1, username: "JohnDoe", avatar: "", bio: "", permid: 1 },
      { id: 2, username: "JaneSmith", avatar: "", bio: "", permid: 1 },
    ])
  ),
}));

describe("PageContent Component Tests", () => {
  test("Renders page information correctly", (done) => {
    const wrapper = shallow(<PageContent match={{ params: { id: 1 } }} />);

    // Wait for the mounted() lifecycle method to complete
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();

      done();
    });
  });

  test("Displays tags correctly", (done) => {
    const wrapper = shallow(<PageContent match={{ params: { id: 1 } }} />);

    setTimeout(() => {
      expect(
        wrapper.containsAllMatchingElements([
          <a className="blue-link" href="/#/tags/1">
            Tag1
          </a>,
          <a className="blue-link" href="/#/tags/2">
            Tag2
          </a>,
        ])
      ).toEqual(true);
      done();
    });
  });

  test("Displays comments correctly", (done) => {
    const wrapper = shallow(<PageContent match={{ params: { id: 1 } }} />);

    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
});
