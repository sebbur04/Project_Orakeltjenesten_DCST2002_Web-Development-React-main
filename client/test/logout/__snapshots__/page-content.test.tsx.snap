// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent Component Tests Displays comments correctly 1`] = `
<div
  className="mainContent"
>
  <Card
    title="Sample Page"
  >
    <Row>
      <Column
        width={12}
      >
        Tagger:
      </Column>
      <Column
        key="1"
        width={1}
      >
        <div
          className="small-text-page-comment"
        >
          <a
            className="blue-link"
            href="/#/tags/1"
          >
            Tag1
          </a>
        </div>
      </Column>
      <Column
        key="2"
        width={1}
      >
        <div
          className="small-text-page-comment"
        >
          <a
            className="blue-link"
            href="/#/tags/2"
          >
            Tag2
          </a>
        </div>
      </Column>
    </Row>
    <div
      className="page-content"
    >
      <Qe>
        ## Sample Content
      </Qe>
    </div>
    <Row>
      <Column
        width={4}
      >
        Antall visninger: 
        123
      </Column>
      <Column
        width={4}
      >
        <a
          className="blue-link"
          href="/#/pages/1/versionlog"
        >
          Versjonslogg
        </a>
      </Column>
    </Row>
  </Card>
  <Card
    title="Kommentarer"
  >
    <div
      className="comment-border"
      key="1"
    >
      <Row>
        <Row>
          <Column>
            <p>
               
              <a
                className="blue-link"
                href="/#/users/1"
              >
                JohnDoe
              </a>
            </p>
          </Column>
          <Column>
            <img
              className="imgstyle"
              src=""
            />
          </Column>
        </Row>
        <Row>
          <div
            className="small-text-page-comment"
          >
            <Column
              width={12}
            >
              2024-01-01 13:30
            </Column>
          </div>
        </Row>
        <Row>
          <br />
        </Row>
         
        <Row
          key="1"
        >
          <Column
            width={10}
          >
            Great post!
          </Column>
        </Row>
      </Row>
    </div>
    <div
      className="comment-border"
      key="2"
    >
      <Row>
        <Row>
          <Column>
            <p>
               
              <a
                className="blue-link"
                href="/#/users/2"
              >
                JaneSmith
              </a>
            </p>
          </Column>
          <Column>
            <img
              className="imgstyle"
              src=""
            />
          </Column>
        </Row>
        <Row>
          <div
            className="small-text-page-comment"
          >
            <Column
              width={12}
            >
              2024-01-01 14:00
            </Column>
          </div>
        </Row>
        <Row>
          <br />
        </Row>
         
        <Row
          key="2"
        >
          <Column
            width={10}
          >
            Needs more details.
          </Column>
        </Row>
      </Row>
    </div>
  </Card>
</div>
`;

exports[`PageContent Component Tests Renders page information correctly 1`] = `
<div
  className="mainContent"
>
  <Card
    title="Sample Page"
  >
    <Row>
      <Column
        width={12}
      >
        Tagger:
      </Column>
      <Column
        key="1"
        width={1}
      >
        <div
          className="small-text-page-comment"
        >
          <a
            className="blue-link"
            href="/#/tags/1"
          >
            Tag1
          </a>
        </div>
      </Column>
      <Column
        key="2"
        width={1}
      >
        <div
          className="small-text-page-comment"
        >
          <a
            className="blue-link"
            href="/#/tags/2"
          >
            Tag2
          </a>
        </div>
      </Column>
    </Row>
    <div
      className="page-content"
    >
      <Qe>
        ## Sample Content
      </Qe>
    </div>
    <Row>
      <Column
        width={4}
      >
        Antall visninger: 
        123
      </Column>
      <Column
        width={4}
      >
        <a
          className="blue-link"
          href="/#/pages/1/versionlog"
        >
          Versjonslogg
        </a>
      </Column>
    </Row>
  </Card>
  <Card
    title="Kommentarer"
  >
    <div
      className="comment-border"
      key="1"
    >
      <Row>
        <Row>
          <Column>
            <p>
               
              <a
                className="blue-link"
                href="/#/users/1"
              >
                JohnDoe
              </a>
            </p>
          </Column>
          <Column>
            <img
              className="imgstyle"
              src=""
            />
          </Column>
        </Row>
        <Row>
          <div
            className="small-text-page-comment"
          >
            <Column
              width={12}
            >
              2024-01-01 13:30
            </Column>
          </div>
        </Row>
        <Row>
          <br />
        </Row>
         
        <Row
          key="1"
        >
          <Column
            width={10}
          >
            Great post!
          </Column>
        </Row>
      </Row>
    </div>
    <div
      className="comment-border"
      key="2"
    >
      <Row>
        <Row>
          <Column>
            <p>
               
              <a
                className="blue-link"
                href="/#/users/2"
              >
                JaneSmith
              </a>
            </p>
          </Column>
          <Column>
            <img
              className="imgstyle"
              src=""
            />
          </Column>
        </Row>
        <Row>
          <div
            className="small-text-page-comment"
          >
            <Column
              width={12}
            >
              2024-01-01 14:00
            </Column>
          </div>
        </Row>
        <Row>
          <br />
        </Row>
         
        <Row
          key="2"
        >
          <Column
            width={10}
          >
            Needs more details.
          </Column>
        </Row>
      </Row>
    </div>
  </Card>
</div>
`;
